keps/sig-storage/3762-persistent-volume-release-timestamp/README.md

```diff
@@ -0,0 +1,878 @@
+-->
+
+### Goals
+
+1) Introduce a new status field in PersistentVolumes.
+2) Update the new field with a timestamp of when the volume was released.
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+### Non-Goals
+
+1) Implement any form of volume health monitoring.
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+We need to update api server to support the newly proposed field and update PV controller to set a value of the new
+timestamp field when it changes volume phase. Also, if the feature gate is disabled the value must be re-set to `nil` 
+when updating or creating a volume.
+
+The value of the field is not intended for use by any other Kubernetes components at this point and should be used only
+as a convenience feature for cluster admins. Cluster admins should be able to list and sort PersistentVolumes based on
+a timestamp which indicates when the volume was moved to a `Released` state.
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a cluster admin I want to use `Retain` policy for released volumes which is safer than `Delete` and gives me more
+control over volume cleanup. However, this in turn leads to a growing number of volumes that need to be monitored
+and cleaned up. The cleanup should be performed based how long the volume was not used by any claim.
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+The new field is purely informative and should not introduce any risk.
+
+## Design Details
+
+Changes required for this KEP:
+
+* kube-apiserver
+  * extend PersistentVolumeStatus type with `ReleaseTimestamp` field
```

Comment:
`LastPhaseTransitionTime`?

Reply:
Sounds good, let's use it in feature gate name too so the naming is consistent.